SteamProject/MySteam/Steam.cpp

```diff
@@ -0,0 +1,261 @@
+#include "Steam.h"
+#include <iostream>
+
+
+void FSteam::RunApp()
+{
+	while (true)
+	{
+		int UserOption;
+		std::cout << "------ Welcome to Steam! ------" << std::endl;
+		std::cout << "\t1 - Add New game \n\t2 - Categories Menu... \n\t3 - Show Games... \n\t4 - Exit \n" << ">>";
+		std::cin >> UserOption;
+
+		switch (UserOption)
+		{
+		case AddGame:
+			OpenAddGameMenu();
+			break;
+
+		case ShowCategories:
+			OpenCategoriesMenu();
+			break;
+
+		case ShowGames:
+			OpenShowGamesMenu();
+			break;
+
+		case Exit:
+			exit(1);
+			break;
+
+		default:
+			std::cin.clear();
+			std::cin.ignore(100, '\n');
+			break;
+		}
+
+		system("cls");
+	}
+}
+
+
+int FSteam::ValidateInput(const std::string Question)
+{
+	bool bRepeat;
+	do
+	{
+		std::cout << Question << " >>";
+		int Input;
+		std::cin >> Input;
+
+		if (std::cin.fail())
+		{
+			bRepeat = true;
+			std::cin.clear();
+			std::cin.ignore(100, '\n');
+			system("cls");
+		}
+		else
+		{
+			bRepeat = false;
+			return Input;
+		}
+
+	} while (bRepeat);
+}
+
+FGame FSteam::AskForNewGame()
+{
+	bool bRepeat;
+	do
+	{
+		system("cls");
+
+		std::cout << "Enter the game name >>";
+		std::string NewGameName;
+		std::cin >> NewGameName;
+
+		std::cout << "Enter the name of the studio that develop the game >>";
+		std::string NewStudioName;
+		std::cin >> NewStudioName;
+
+		int GameDay = ValidateInput("Enter the release day");
+		int GameMonth = ValidateInput("Enter the release month");
+		int GameYear = ValidateInput("Enter the release year");
+
+		FGame NewGame = FGame(NewGameName, NewStudioName, GameDay, GameMonth, GameYear);
+
+		if (NewGame.IsValid())
+		{
+			bRepeat = false;
+			return NewGame;
+		}
+		else
+		{
+			bRepeat = true;
+		}
+
+	} while (bRepeat);
+}
+
+void FSteam::OpenAddGameMenu()
+{
+	FGame NewGame = AskForNewGame();
+
+	bool bGameAdded;
+	char UserChoice;
+	do
+	{
+		system("cls");
+		std::cout << "Do you want to add the game to an existing category?(y/n)\n" << ">>";
+		std::cin >> UserChoice;
+
+		if (MainHandler.GetCategoriesAmount() <= 0)
+		{
+			UserChoice = 'n';
```

Comment:
Remember that we've discussed this. It's confusing to take variables that are supposed to represent one thing and set them as other things. 

The variable UserChoice is supposed to represent the user choice. So it is very confusing to ask the user for their input and then if categories are empty, change their answer as if they had answered `n` regardless of what they answered. And also later on, setting their choice as 'r' when that wasn't even an option.

Until I read your whole function, I have no idea what the while condition means.

Reply:
What you could do instead is divide this into functions. If you want, we can discuss this tomorrow better.